tests/tests_test_workflow/test_dependency_installer.py

```diff
@@ -13,16 +13,26 @@ class DependencyInstallerTests(unittest.TestCase):
     DIST_MANIFEST_LOCAL = os.path.join(DATA, "local", "dist", "manifest.yml")
     DIST_MANIFEST_REMOTE = os.path.join(DATA, "remote", "dist", "manifest.yml")
 
+    @patch("concurrent.futures.ThreadPoolExecutor", return_value=MagicMock())
     @patch("os.makedirs")
     @patch("shutil.copyfile")
     @patch("urllib.request.urlretrieve")
-    def test_install_maven_dependencies_local(self, mock_request, mock_copyfile, mock_makedirs):
+    def test_install_maven_dependencies_local(self, mock_request, mock_copyfile, mock_makedirs, mock_threadpool):
+        def submit_and_run(callable, source, dest):
+            # Filter down the actual calls to make analyzing failures easier
+            if 'alerting-notification-1.2.0.0.jar' in source:
+                return callable(source, dest)
+            return None
+        mock_threadpool.return_value.__enter__().submit.side_effect = submit_and_run
         dependency_installer = DependencyInstaller(
             self.DATA,
             BuildManifest.from_path(self.BUILD_MANIFEST),
             BundleManifest.from_path(self.DIST_MANIFEST_LOCAL)
         )
         dependency_installer.install_maven_dependencies()
+        self.assertEqual(mock_threadpool.call_count, 4)
+        self.assertEqual(mock_threadpool().__enter__().submit.call_count, 2375)
```

Comment:
Can you explain how is submit call count 2375? Never used threadpools before

Reply:
that `submit(...)` function is called 2375 times, for each file in the manifest.  This assert is making sure that we actually performed the action that number of times.  If something was wrong with the executor it could record more/less and we would want to know

Reply:
Got it. Thanks!